interface_tester/interface_test.py

```diff
@@ -410,19 +413,27 @@ def _run_scenario(self, event: Event, state: State):
         )
         return ctx.run(event, state)
 
-    def _cast_event(self, raw_event: Union[str, Event], relation: Relation):
-        # test.EVENT might be a string or an Event. Cast to Event.
-        event = Event(raw_event) if isinstance(raw_event, str) else raw_event
-
-        if not isinstance(event, Event):
-            raise InvalidTestCaseError(
-                f"Expected Event or str, not {type(raw_event)}. "
-                f"Invalid test case: {self} cannot cast {raw_event} to Event."
-            )
+    def _cast_event(self, raw_event: Union[str, _Event], relation: Relation):
+        if isinstance(raw_event, str):
+            if raw_event.endswith("-relation-changed"):
+                event = CharmEvents.relation_changed(relation)
+            elif raw_event.endswith("-relation-departed"):
+                event = CharmEvents.relation_departed(relation)
+            elif raw_event.endswith("-relation-broken"):
+                event = CharmEvents.relation_broken(relation)
+            elif raw_event.endswith("-relation-joined"):
+                event = CharmEvents.relation_joined(relation)
+            elif raw_event.endswith("-relation-created"):
+                event = CharmEvents.relation_created(relation)
+            else:
+                raise InvalidTestCaseError(
+                    f"Bad interface test specification: event {raw_event} is not a relation event."
+                )
 
-        if not event._is_relation_event:
+        if not isinstance(event, _Event):
```

Comment:
ehm, what is "event" now referring to?

Reply:
and why don't we do this type checking earlier?

Reply:
> ehm, what is "event" now referring to?

Yeah, missed fixing the name there, sorry. Fixed now.

> and why don't we do this type checking earlier?

It felt cleaner to do the "if string then do this" first and then "if not _Event error" rather than "if not str or _Event error" followed by "if string". I've changed it back to the other order.